src/governance-classes.cpp

```diff
@@ -546,6 +546,27 @@ CAmount CSuperblock::GetPaymentsLimit(int nBlockHeight)
     return nPaymentsLimit;
 }
 
+void CSuperblock::GetSuperblockHeight(int& nBlockHeight, int& nLastSuperblock, int& nNextSuperblock)
+{
+    // Get current block height if nothing was provided
+    if(nBlockHeight == 0)
+        nBlockHeight = (int)chainActive.Height();
+
+    // Special case: return first superblock if the current block height is below the first superblock during sync
+    if(nBlockHeight < Params().GetConsensus().nSuperblockStartBlock){
+        nLastSuperblock = 0;
+        nNextSuperblock = Params().GetConsensus().nSuperblockStartBlock;
+        return;
+    }
+
+    // Get last superblock
+    nLastSuperblock = nBlockHeight - ((nBlockHeight - Params().GetConsensus().nSuperblockStartBlock) % Params().GetConsensus().nSuperblockCycle);
```

Comment:
yep, @crowning- take a look at `CSuperblock::IsValidBlockHeight()`, `nSuperblockStartBlock` is more like a "checkpoint" after which superblocks are possible at all, not "the first superblock".